internal/pkg/cli/dialog/create_workspace.go

```diff
@@ -0,0 +1,105 @@
+package dialog
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/keboola/go-client/pkg/sandbox"
+	"github.com/keboola/keboola-as-code/internal/pkg/cli/options"
+	"github.com/keboola/keboola-as-code/internal/pkg/cli/prompt"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils"
+	workspace "github.com/keboola/keboola-as-code/pkg/lib/operation/project/remote/workspace"
+)
+
+type createWorkspaceDeps interface {
+	Options() *options.Options
+}
+
+func (p *Dialogs) AskCreateWorkspace(d createWorkspaceDeps) (workspace.CreateOptions, error) {
+	opts := workspace.CreateOptions{}
+
+	name, err := p.askWorkspaceName(d)
+	if err != nil {
+		return opts, err
+	}
+	opts.Name = name
+
+	typ, err := p.askWorkspaceType(d)
+	if err != nil {
+		return opts, err
+	}
+	opts.Type = typ
+
+	size, err := p.askWorkspaceSize(d)
+	if err != nil {
+		return opts, err
+	}
+	opts.Size = size
+
+	return opts, nil
+}
+
+func (p *Dialogs) askWorkspaceName(d createWorkspaceDeps) (string, error) {
+	if d.Options().IsSet("name") {
+		return d.Options().GetString("name"), nil
+	} else {
+		name, ok := p.Ask(&prompt.Question{
+			Label:     "Enter a name for the new workspace",
+			Validator: prompt.ValueRequired,
+		})
+		if !ok || len(name) == 0 {
+			return "", fmt.Errorf("missing name, please specify it")
+		}
+		return name, nil
+	}
+}
+
+var workspaceTypes = []string{
+	"python",
+	"snowflake",
+	"r",
+}
+
+func (p *Dialogs) askWorkspaceType(d createWorkspaceDeps) (string, error) {
+	if d.Options().IsSet("type") {
+		typ := d.Options().GetString("type")
+		if !utils.Contains(workspaceTypes, typ) {
+			return "", fmt.Errorf("invalid workspace type, must be one of: %s", strings.Join(workspaceTypes, ", "))
+		}
+		return typ, nil
+	} else {
+		v, ok := p.Select(&prompt.Select{
+			Label:   "Select a type for the new workspace",
+			Options: workspaceTypes,
+		})
+		if !ok {
+			return "", fmt.Errorf("missing workspace type, please specify it")
+		}
+		return v, nil
+	}
+}
+
+var workspaceSizes = []string{
+	sandbox.SizeSmall,
+	sandbox.SizeMedium,
+	sandbox.SizeLarge,
+}
```

Comment:
Hodis to prosim do `go-client`? Vdaka.

Reply:
BTW, ked z toho spravis `map[Size]bool`, tak budes vediet vyhladat hodnotu s `O(1)` a bez `utils.Contains`.

Reply:
> zase nemá literal syntax
tomu nerozumiem
![image](https://user-images.githubusercontent.com/19371734/192270104-255ee631-c94f-4ce9-a745-d2cc2ef0559f.png)
https://go.dev/play/p/-BQ1r9r1Dx6

Reply:
aha, beriem spat, ok `orderedmap`, nie `map`

Reply:
Mozu to byt aj 2 hodnoty: `sandbox.SizesOrdered`, `sandbox.SizesMap`

Reply:
Samozrejme, je to drobnost, takze to necham na teba, ake riesenie si z vyberies.

Reply:
Bolo by to dobre aspon presunut, a pouzit existujucu funkciu https://pkg.go.dev/golang.org/x/exp/slices#Contains

Osobne by som preferoval tie 2 hodnoty.

Reply:
Dobře, rozdělím to na dvě hodnoty, které dám do `go-client` a odstraním `utils.Contains`